lib/paypal_express/models/response.rb

```diff
@@ -77,6 +77,22 @@ def self.initial_payment_account_processor_id(kb_account_id, kb_tenant_id, token
         response.nil? ? nil : response.payment_processor_account_id
       end
 
+      def self.update_response_for_pending_payment(transaction_plugin_info)
+        response = where(:api_call => 'build_form_descriptor',
+                         :kb_payment_id => transaction_plugin_info.kb_payment_id,
+                         :kb_payment_transaction_id => transaction_plugin_info.kb_transaction_payment_id).last
+        return nil if response.nil?
+        response.update_pending_to_cancel
```

Comment:
Could you simply do `where().update_all()`? Hopefully, a single transaction would be created in that case?

Might be relevant: http://stackoverflow.com/questions/9186822/how-do-i-do-an-update-statement-where-a-condition.